qubes/ext/block.py

```diff
@@ -189,10 +190,41 @@ def parent_device(self) -> Optional[qubes.devices.Device]:
                 self._interface_num = interface_num
         return self._parent
 
+    @property
+    def attachment(self) -> Optional['qubes.vm.BaseVM']:
+        """
+        Warning: this property is time-consuming, do not run in loop!
+        """
+        if not self.backend_domain.is_running():
+            return None
+        for vm in self.backend_domain.app.domains:
+            if not vm.is_running():
+                continue
+            if self._is_attached_to(vm):
+                return vm
+
+        return None
```

Comment:
Should there be an index to make looking this up fast?

Reply:
Probably, but we need to remember that it should always be up to date (not cached), if you have an idea how to do it, I'm in.

Reply:
Updating it whenever a device is attached or detached is the first idea that comes to mind.  Caches are fine so long as they are invalidated synchronously when needed.

Reply:
`BlockDeviceExtension` already has a `devices_cache`, but `attachment` is intended to return the actual state (including, in particular, situations where something went wrong and the potential cache is outdated). For API compatibility, this is a `property`, and the warning is there because when we want to get information about `attachment` for a bunch of devices, we should use some form of bulk data loading (just like we do for updating `devices_cache`) rather than in a loop.

Reply:
Are there any situations where the cache could be outdated, or is that always a bug?

Reply:
Not sure, but if it is predictable it's already update a cache